main.go

```diff
@@ -527,6 +583,68 @@ func DeleteS3Backups(backupTime time.Time, retentionPeriod time.Duration, svc *m
 	}
 }
 
+func DeleteBackupAction(c *cli.Context) error {
+	name := c.String("name")
+	if name == "" {
+		return fmt.Errorf("snapshot name is required")
+	}
+	compressedPath := fmt.Sprintf("/backup/%s.%s", name, compressedExtension)
+	uncompressedPath := fmt.Sprintf("/backup/%s", name)
+
+	// Since we have to support compressed and uncompressed versions of snapshots.
+	// We can't remove the uncompressed snapshot during cleanup unless we are
+	// sure the compressed is there, hence the complex check.
+	if c.Bool("cleanup") {
+		if _, err := os.Stat(compressedPath); err == nil {
+			// for cleanup, we only want to delete the uncompressed snapshot.
+			// we don't need to go to s3
+			return deleteBackup(uncompressedPath)
+		}
+	} else {
```

Comment:
Can this else be dropped as we return in the `c.Bool("cleanup")`?

Reply:
No, `c.Bool("cleanup")` has another condition to return, if that's not met, we don't want to run the code in the else block.

Reply:
got it